rpc/grpc/client/block_service.go

```diff
@@ -0,0 +1,123 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	blocksvc "github.com/cometbft/cometbft/proto/tendermint/services/block/v1"
+	"github.com/cometbft/cometbft/types"
+	"github.com/cosmos/gogoproto/grpc"
+)
+
+// Block data returned by the CometBFT BlockService gRPC API.
+type Block struct {
+	BlockID types.BlockID `json:"block_id"`
+	Block   *types.Block  `json:"block"`
+}
+
+// LatestHeightResult type used in GetLatestResult and send to the client
+// via a channel
+type LatestHeightResult struct {
+	Height int64
+	Error  error
+}
+
+// BlockServiceClient provides block information
+type BlockServiceClient interface {
+	GetBlockByHeight(ctx context.Context, height int64) (*Block, error)
```

Comment:
Is is better for `GetBlockByHeight` on the client interface to take an `int64` height and then instantiate a `GetByHeightRequest` in method scope?

Reply:
Yes. This particular client wrapper interface should be as trivially ergonomic as possible. Just like in tendermint-rs, this involves "unwrapping" the underlying RPC request and response structures on behalf of the callers.

If callers want to use the underlying gRPC client directly, they're also welcome to do that.

Reply:
Yes, this client wrapper interface should be as trivially ergonomic as possible. Just like in tendermint-rs' RPC client, this involves "unwrapping" key request struct members into function parameters (and similarly for responses).

Callers are always welcome to either generate their own gRPC clients, or use our generated one directly if they need access to them for some reason.